internal/controller/postgrescluster/pgbouncer_test.go

```diff
@@ -365,3 +371,87 @@ func TestReconcilePGBouncerDeployment(t *testing.T) {
 	})
 
 }
+
+func TestReconcilePGBouncerDisruptionBudget(t *testing.T) {
+	env, cc, config := setupTestEnv(t, ControllerName)
+	t.Cleanup(func() { teardownTestEnv(t, env) })
+
+	r := &Reconciler{}
+	ctx, cancel := setupManager(t, config, func(mgr manager.Manager) {
+		r = &Reconciler{
+			Client:   cc,
+			Owner:    client.FieldOwner(t.Name()),
+			Recorder: mgr.GetEventRecorderFor(ControllerName),
+			Tracer:   otel.Tracer(t.Name()),
+		}
+	})
+	t.Cleanup(func() { teardownManager(cancel, t) })
+
+	foundPDB := func(
+		t *testing.T,
+		cluster *v1beta1.PostgresCluster,
+		minAvailable int32,
+	) bool {
+		pdb, err := r.generatePodDisruptionBudget(
+			cluster,
+			naming.ClusterPGBouncer(cluster),
+			minAvailable,
+			naming.ClusterPGBouncerSelector(cluster),
+		)
+		assert.NilError(t, err)
+		got := &policyv1beta1.PodDisruptionBudget{}
+		err = r.Client.Get(ctx, client.ObjectKeyFromObject(pdb), got)
+		return !apierrors.IsNotFound(err)
+	}
+
+	ns := &corev1.Namespace{}
+	ns.GenerateName = "postgres-operator-test-"
+	ns.Labels = labels.Set{"postgres-operator-test": t.Name()}
+	assert.NilError(t, cc.Create(ctx, ns))
+	t.Cleanup(func() { assert.Check(t, cc.Delete(ctx, ns)) })
+
+	t.Run("empty", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns.Name
+		cluster.Spec.Proxy = nil
+
+		assert.NilError(t, r.reconcilePGBouncerPodDisruptionBudget(ctx, cluster))
+	})
+
+	t.Run("no replicas in spec", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns.Name
+		cluster.Spec.Proxy.PGBouncer.Replicas = nil
+		assert.Equal(t, r.reconcilePGBouncerPodDisruptionBudget(ctx, cluster).Error(),
+			"Replicas should be defined")
+	})
+
+	t.Run("not created", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns.Name
+		cluster.Spec.Proxy.PGBouncer.Replicas = initialize.Int32(1)
+		cluster.Spec.Proxy.PGBouncer.MinAvailable = &intstr.IntOrString{IntVal: int32(0)}
+		assert.NilError(t, r.reconcilePGBouncerPodDisruptionBudget(ctx, cluster))
+		assert.Assert(t, !foundPDB(t, cluster, 0))
+	})
+
+	t.Run("created", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.UID = types.UID("hippouid")
```

Comment:
🤔 This UID is for the controller reference, right? I think this test races with the garbage collector in a full Kubernetes cluster. Better to call `cc.Create(ctx, cluster)` without a Manager.